### ISSUE TITLE
Adapt test cases to use repeatable @Mappings and @ValueMappings

### ISSUE THREAD

**Issue by filiphr** [mapstruct/mapstruct#1628]
We should still keep some usages of `@Mappings` and `@ValueMappings` in order to just validate that it works properly.

Issue coming from https://github.com/mapstruct/mapstruct/pull/1623#pullrequestreview-164473536

**Comment by sjaakd** [comment]
> We should still keep some usages of `@Mappings` and `@ValueMappings` in order to just validate that it works properly.

Agreed. Do take into consideration that I ran into trouble when using `@Repeatable` in #1618 (4ccc875). Seems that errors in Tycho / JDT are reported on the method in stead of the offending `@Mapping` annotation.

**Comment by filiphr** [comment]
Migrating to this is not really an option at the moment due to a bug in the java compiler. Once [JDK-8042710](https://bugs.openjdk.java.net/browse/JDK-8042710) is resolve we can do this
